### DISCUSSION TITLE
Advance Payment Invoices

### DISCUSSION THREAD

**Discussion by BOTSolutions8686** [lavaloon-eg/ksa_compliance#136] [Q&A] [answered]
Hi . I would like to know if the module handles advance payment invoice. 
I have clients from construction business and their invoices usually contain the following fields 
- Retention
- Deduction 
- Recovery of advance payment

In case of recovery of advance payment, previous invoice number is to be referred on the invoice to be submitted and the amount is also deducted from the total of the current invoice. 
How can this be handled with the module or if it automatically handles this. 
any insight on this will be much appreciated. 
Thank you.

**Comment by mhaggag** [comment]
Hello,

Pre-payment is not supported yet. I'm not sure whether your scenario can be modeled/implemented without that support. @omarashrafsarhan Please advise.

**Reply by BOTSolutions8686** [reply]
Prior to this we were using the module provided by ERPGulf . We had to make a lot of modifications for Advance payment invoices in the erpnext. It solved the problem of submitting the invoice which was cleared by ZATCA. But made the accounting of erpnext practically useless. Especially when introducing "Retention" Deduction" and "Recovery of Advance" 
As recovery of advance needed a reference from an old invoice Which we brought into the current invoice using child table and the user was able to select from a list of Advance payment invoices to refer and the amount that is to be recovered. 

Your module on the other hand is very very stable for all types of transactions. I was hoping if you can look into advance payment invoices, the solution that you will produce will be much more stable and usable. 

Thank you

**Reply by BOTSolutions8686** [reply]
Some screens that might help 
<img width="1220" alt="Screenshot 2024-10-05 at 7 22 35 PM" src="https://github.com/user-attachments/assets/10964bd9-d8b2-4c36-8c16-7e7a979d64ea">
<img width="1215" alt="Screenshot 2024-10-05 at 7 22 16 PM" src="https://github.com/user-attachments/assets/5c718148-0145-40ca-8764-456e6ebae79f">
<img width="1216" alt="Screenshot 2024-10-05 at 7 21 53 PM" src="https://github.com/user-attachments/assets/c6b142d2-1635-4a5c-85b8-b8ad97cc794a">

**Reply by omarashrafsarhan** [reply, marked as answer]
Hello,
Prepayment invoice type is indeed part of our roadmap, our current focus is on enhancing app stability and improving the user experience for the main invoice types, specifically TAX invoices and simplified invoices.

However, we are always attentive to the needs and feedback of the community. Should there be significant interest in the prepayment invoice type, we would certainly consider adjusting our priorities accordingly.

Please feel free to share any further thoughts or requirements you might have.

**Reply by BOTSolutions8686** [reply]
Thank you . 
Really looking forward to this update. 
And appreciate your response and attention to all queries.
